controllers/nnf_workflow_controller_helpers.go

```diff
@@ -996,35 +1002,30 @@ func (r *NnfWorkflowReconciler) createOrUpdateContainerServiceIfNecessary(ctx co
 		},
 	}
 
-	mutateFn := func() error {
-		service.Spec.Selector = map[string]string{
-			nnfv1alpha1.ContainerLabel: workflow.Name,
-		}
+	service.Spec.Selector = map[string]string{
+		nnfv1alpha1.ContainerLabel: workflow.Name,
+	}
 
-		service.Spec.ClusterIP = corev1.ClusterIPNone
+	service.Spec.ClusterIP = corev1.ClusterIPNone
 
-		if err := ctrl.SetControllerReference(workflow, service, r.Scheme); err != nil {
-			return fmt.Errorf("setting Service controller reference failed for '%s': %w", service.Name, err)
-		}
-
-		return nil
+	if err := ctrl.SetControllerReference(workflow, service, r.Scheme); err != nil {
+		return fmt.Errorf("setting Service controller reference failed for '%s': %w", service.Name, err)
 	}
 
-	result, err := ctrl.CreateOrUpdate(ctx, r.Client, service, mutateFn)
+	err := r.Create(ctx, service)
 	if err != nil {
-		return err
+		if !apierrors.IsAlreadyExists(err) {
+			return err
+		}
+		log.Info("Service already exists", "object", client.ObjectKeyFromObject(service).String())
```

Comment:
Is this log helpful? We should try to keep things lean. Maybe log.Debug()?

Reply:
We only have Info() and Error(). I thought Info() was better than nothing, but I can take it out.

Reply:
That's right, logr supports V() levels and not Debug. We don't have well-defined levels in our code, and I think now is not the time to introduce it.

But, as to the usefulness, it helps me to ask myself "Am I learning anything from the log statement?" Otherwise it's just noise.